### ISSUE TITLE
`avoid_redundant_argument_values` is not working for enum instances

### ISSUE THREAD

**Issue by AlexV525** [dart-lang/linter#3617]
```dart
enum TestEnum {
  a(test: true),
  b(test: false); // <-- Should report positive.

  const TestEnum({this.test = false});

  final bool test;
}
```

In comparison, a working case with classes:

```dart
class TestClass {
  const TestClass({this.test = false});

  final bool test;

  static const a = TestClass(test: true);
  static const b = TestClass(test: false); // <-- Report correctly.
}
```

**Comment by AlexV525** [comment]
This is blocked by the new `visitEnumConstantArguments` in the analyzer package.

**Comment by bwilkerson** [comment]
Blocked in what way?

**Comment by AlexV525** [comment]
> Blocked in what way?

None. :) Submitted a PR.

**Comment by AlexV525** [comment]
The PR uses `visitEnumDeclaration` at present, and I think it's better to use `visitEnumConstantArguments`, which is still blocked by dart-lang/sdk#49821.
